src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -752,6 +791,96 @@ protected function compileEndpush($expression)
     }
 
     /**
+     * Compile the inject annotations into valid PHP.
+     *
+     * @param  string[] $segments
+     *
+     * @return string
+     */
+    protected function compileInjectAnnotation(array $segments)
+    {
+        list($injectClass, $variableName) = $segments;
+
+        return '<?php '.$variableName.' = app('.$injectClass.'::class); ?>';
```

Comment:
The reason for this is so it works correctly with `@use`. If a string is used as you say aliases or imported classes will not be found. This would be really confusing given the syntax has the type as an identifier. Why is it 'dangerous'?